CMakeLists.txt

```diff
@@ -11,7 +11,8 @@ find_package(Python COMPONENTS Interpreter Development REQUIRED)
 Python_add_library(_nvjitlinklib MODULE pynvjitlink/_nvjitlinklib.cpp WITH_SOABI)
 
 find_package(
-  CUDAToolkit 12.3 REQUIRED
+  # Require CUDA 12.2 Update 2 to avoid nvjitlink bugs
+  CUDAToolkit 12.2.140 REQUIRED
```

Comment:
Why does this move backwards from 12.3 to 12.2? I have read the discussions on the PR, but it looked like the conclusions suggested the latest was necessary. (FWIW, my POV has always been that the optimal thing to do is to always build against the latest released version)

Reply:
If the user has a 12.3 toolkit and 12.2 driver then it wouldn't work. A couple of other references from comments on this PR:

* https://github.com/rapidsai/pynvjitlink/pull/33#discussion_r1446404877
* https://github.com/rapidsai/pynvjitlink/pull/33#discussion_r1446523241

Reply:
In CUDA 12, the entire CUDA Toolkit is shipped via Conda and there are no CUDA 12.3 Conda packages currently. So that doesn't seem like a risk

We can of course add a `cuda-version` upper bound to guard against this ( https://github.com/rapidsai/pynvjitlink/pull/33/files#r1445457742 )

Reply:
The other thing to keep in mind is there will always be newer CUDA Toolkits released after we have published RAPIDS releases (especially when considering LTS releases). So we cannot guarantee users will have a version of `pynvjitlink` that was built against the just recently shipped CUDA Toolkit

Think the best we can do in those cases is have proper version constraints and maybe do some validation on `import pynvjitlink` to catch any other aberrant cases

Reply:
> In CUDA 12, the entire CUDA Toolkit is shipped via Conda and there are no CUDA 12.3 Conda packages currently. So that doesn't seem like a risk

The user could install Numba with conda, not install the CUDA toolkit at all via conda, and use a system-installed 12.3 toolkit - you don't need to conda-install a CUDA toolkit to use it with Numba.

Reply:
At least today, that is not a supported use case of CUDA with Conda in RAPIDS or with conda-forge packages

Reply:
@gmarkall We discussed this topic a bit on Slack. I'll tag you there. This PR implements a "short term plan" by using conda-forge's pynvjitlink from 12.2u2 so we can update RAPIDS to build with CUDA 12.2. We've also discussed a "long term plan" to use tarballs or some other upstream source that reliably provides "latest CUDA" that we can use as a single source for both conda and pip packages. That will align more closely with what you're seeking. For this PR, we just need to identify if this is good enough for a first iteration of pynvjitlink conda packages. This is needed to unblock cudf building with CUDA 12.2, since pynvjitlink will be a hard dependency of cudf.